src/charm.py

```diff
@@ -28,26 +28,42 @@ def __init__(self, *args):
             log.info("Not a leader, skipping set_pod_spec")
             self.model.unit.status = ActiveStatus()
             return
+
         self._stored.set_default(secret_key=gen_pass())
-        self.minio_interface = MinioProvide(self, "minio")
+        try:
+            self.interfaces = get_interfaces(self)
+        except NoVersionsListed:
+            self.model.unit.status = WaitingStatus("Waiting for version handshake")
```

Comment:
I'm not sure if this is the most descriptive message we can give here. Maybe we could somehow mention that this is something relations related.

Reply:
Now uses the error message generated by SDI